### PR TITLE
Fix bug in Analysis report generation

### PR DESCRIPTION
**Pull request by lucernae**
- Allow easier Analysis Rerun in detail page (for developer only)
- Fix Impact Report location resolver to remember report

### CODE REVIEW

**Comment by lucernae** [comment]
Depends on https://github.com/inasafe/inasafe-headless/pull/49

File: tasks/analysis.py [hunk 1]
```diff
@@ -628,15 +628,27 @@ def process_impact_report(analysis, report_metadata):
         # TODO: find out how to upload document using post request
 
         assign_report = {
-            'map-report': analysis.assign_report_map,
+            # By default we always take the portrait report
+            'map-report-portrait': analysis.assign_report_map,
```
Comment (myarjunar) [inline tasks/analysis.py:632]: I just realized that the default key is `inasafe-map-report-portrait`,  we should change it

**Review by myarjunar — APPROVED** [review]
my bad here, `map-report-portrait` was not the correct report key from inasafe.

File: tasks/analysis.py (hunk 1 above)
Comment (myarjunar) [reply tasks/analysis.py:632]:
```suggestion
            'inasafe-map-report-portrait': analysis.assign_report_map,
```

File: tasks/analysis.py [hunk 2]
```diff
@@ -628,15 +628,27 @@ def process_impact_report(analysis, report_metadata):
...
-            if report_exists:
-                assign_report[key](report_metadata['pdf_product_tag'][key])
+        # List of tags of PDF report product
+        pdf_product_tag = report_metadata['pdf_product_tag']
+        for metadata_key in pdf_product_tag.keys():
+            for assign_key, assign_method in assign_report.iteritems():
+
+                # If the key we search doesn't exists in metadata,
+                # just continue
+                if assign_key not in metadata_key:
```
Comment (myarjunar) [inline tasks/analysis.py:642]: ah we are good then. LGTM.

Comment (lucernae) [reply tasks/analysis.py:642]: I thought the key was left intentional to 'map-report-portrait' by you, so I used this logic :D.
Is it better to use 'inasafe-map-report-portrait' or just 'map-report-portrait' ?

**Review by lucernae — COMMENTED** [review]
@myarjunar see my comment above

File: tasks/analysis.py (hunk 1 above)
Comment (lucernae) [reply tasks/analysis.py:632]: The strange thing is, it doesn't get caught in the unittest (or was it?). Maybe it was this line all along.

**Comment by lucernae** [comment]
I have no idea why it still an error, but I will merge it for now.

**Pull request merged**